Source/Plugins/Processing/ProcessingFilters/MinSize.cpp

```diff
@@ -404,6 +409,14 @@ void MinSize::assign_badpoints()
     }
     QString attrMatName = m_FeatureIdsArrayPath.getAttributeMatrixName();
     QList<QString> voxelArrayNames = m->getAttributeMatrix(attrMatName)->getAttributeArrayNames();
+    for(const auto& dataArrayPath : m_IgnoredDataArrayPaths)
+    {
+      if(voxelArrayNames.contains(dataArrayPath.getDataArrayName()))
+      {
+        voxelArrayNames.removeAll(dataArrayPath.getDataArrayName());
```

Comment:
contains(const T&) needs to iterate over all items at worst to return that it doesn't need to change the list.  There shouldn't be a performance loss if there is no change, but there will be a performance boost if even one item needs to be removed.

Reply:
Good catch. Fixed in latest push.